### PR TITLE
whisper-cpp: Add version 1.2.1

### LINKED ISSUE #4616: [Request]: whisper.cpp
### Prerequisites

- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.

### Criteria

- [X] Non-GUI tool
- [X] Reasonably well-known and widely used (e.g. if it's a GitHub project, it should have at least 500 stars and/or 150 forks)
- [X] English interface (or at least English documentation)
- [X] Latest stable version
- [X] Full version (i.e. not a trial version)
- [X] Fairly standard install (e.g. uses a version-specific download URL, no elaborate pre/post install scripts)

### Name

whisper-cpp (or whisper.cpp?)

### Description

High-performance inference of OpenAI's Whisper automatic speech recognition (ASR) model

### Homepage

https://github.com/ggerganov/whisper.cpp

### Download Link(s)

https://github.com/ggerganov/whisper.cpp/releases

### Some Indication of Popularity/Repute

13.8k stars, 1.1k forks

### Note

Model Weights can be separately downloaded here: https://huggingface.co/ggerganov/whisper.cpp
This should likely be mentioned in the package note, as they are required to actually use the application.

### PR DESCRIPTION
**Pull request by leumasme**
Adds [whisper-cpp](https://github.com/ggerganov/whisper.cpp)

Closes #4616

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/whisper-cpp.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        "Whisper Model files are not included. You can download them here:",
+        "https://huggingface.co/ggerganov/whisper.cpp/tree/main"
+    ],
+    "bin": [
+        ["bench.exe", "whisper-bench"],
+        ["command.exe", "whisper-command"],
+        ["main.exe", "whisper"],
+        ["stream.exe", "whisper-stream"],
+        ["talk.exe", "whisper-talk"]
+    ],
```
Comment (rashil2000) [inline bucket/whisper-cpp.json:16]: Is renaming really necessary? I'd prefer to leave them as is (except main.exe).

Comment (leumasme) [reply bucket/whisper-cpp.json:16]: I assumed that the original names would be too common, but we could just leave the names at their default and let the user resolve any name conflicts themselves.

Comment (rashil2000) [reply bucket/whisper-cpp.json:16]: Yes, that's what I was thinking too...

Comment (leumasme) [reply bucket/whisper-cpp.json]: It has been done

File: bucket/whisper-cpp.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "homepage": "https://github.com/ggerganov/whisper.cpp",
+    "license": "MIT",
+    "notes": [
+        "Whisper Model files are not included. You can download them here:",
+        "https://huggingface.co/ggerganov/whisper.cpp/tree/main"
+    ],
+    "bin": [
+        "bench.exe",
+        "command.exe",
+        ["main.exe", "whisper"],
```
Comment (rashil2000) [inline bucket/whisper-cpp.json:13]:
```suggestion
        [
            "main.exe",
            "whisper"
        ],
```

**Pull request merged**